File: typescript/openai-sdk.ts
/**
 * Simple chat completion using the official OpenAI Node SDK.
 * Showcases model usage via the openai package.
 */

import OpenAI from "openai";

const MODEL = "gpt-4o-mini";

function getClient(): OpenAI {
  const apiKey = process.env.OPENAI_API_KEY;
  if (!apiKey) {
    throw new Error("OPENAI_API_KEY is not set");
  }
  return new OpenAI({ apiKey });
}

export async function chat(messages: OpenAI.Chat.ChatCompletionMessageParam[]): Promise<string> {
  const client = getClient();
  const completion = await client.chat.completions.create({
    model: MODEL,
    messages,
    max_tokens: 1024,
    temperature: 0.7,
  });
  const content = completion.choices[0]?.message?.content;
  if (content == null) {
    throw new Error("No content in completion");
  }
  return content;
}

// Run a single prompt when executed directly
async function main(): Promise<void> {
  try {
    const reply = await chat([{ role: "user", content: "Say hello in one sentence." }]);
    console.log(reply);
  } catch (err) {
    console.error(err);
    process.exit(1);
  }
}

main();

File: java/src/main/java/aibuffet/OpenAiSdk.java
package aibuffet;

import com.openai.client.OpenAIClient;
import com.openai.client.okhttp.OpenAIOkHttpClient;
import com.openai.models.chat.completions.ChatCompletion;
import com.openai.models.chat.completions.ChatCompletionCreateParams;

/**
 * Simple chat completion using the official OpenAI Java SDK.
 * Showcases model usage via the openai-java package.
 */
public final class OpenAiSdk {

    private static final String MODEL = "gpt-4o-mini";

    private static OpenAIClient getClient() {
        String apiKey = System.getenv("OPENAI_API_KEY");
        if (apiKey == null || apiKey.isBlank()) {
            throw new IllegalStateException("OPENAI_API_KEY is not set");
        }
        return OpenAIOkHttpClient.builder().apiKey(apiKey).build();
    }

    public static String chat(String userMessage) {
        OpenAIClient client = getClient();
        ChatCompletionCreateParams params = ChatCompletionCreateParams.builder()
                .model(MODEL)
                .maxCompletionTokens(1024L)
                .addUserMessage(userMessage)
                .temperature(0.7)
                .build();

        ChatCompletion completion = client.chat().completions().create(params);
        return completion.choices().stream()
                .findFirst()
                .flatMap(choice -> choice.message().content())
                .orElseThrow(() -> new IllegalStateException("No content in completion"));
    }

    public static void main(String[] args) {
        try {
            System.out.println(chat("Say hello in one sentence."));
        } catch (Exception err) {
            System.err.println(err.getMessage());
            System.exit(1);
        }
    }
}
